docs/source/app_developers_guide/adding_a_node.rst

```diff
@@ -0,0 +1,443 @@
+
+#. If you need to stop or reset the environment, enter CTRL-c in each
+   container to stop the Sawtooth component, then run the following command
+   from your host system:
+
+   .. code-block:: console
+
+      user@host$ docker-compose -f sawtooth-default-poet.yaml down
+
+
+Ubuntu: Configure and Start a Multiple-node Sawtooth Network
+=============================================================
+
+
+Prerequisites
+-------------
+
+This procedure assumes that you have set up an application development
+environment as described in :doc:`ubuntu`, and that you have a working
+(runnable) validator node.
+
+Step 1: Configure the Network on the First Node
+-----------------------------------------------
+
+#. If the first validator node is running, stop all Sawtooth components as
+   described in :ref:`stop-sawtooth-ubuntu-label`.
+
+#. Open a new terminal window on the first validator node (the client-1 window)
+   and configure the Sawtooth settings to allow transactions from the Settings,
+   IntegerKey, XO, and Validator Registry transaction families. Run this
+   command:
+
+     .. code-block:: console
+
+        user@client-1$ sawset proposal create \
+        sawtooth.validator.transaction_families='[{"family": "intkey", "version": "1.0"},{"family":"sawtooth_settings", "version":"1.0"},{"family":"xo", "version":"1.0"},{"family":"sawtooth_validator_registry", "version":"1.0"}]'
+
+#. Create the ``validator.toml`` configuration file on the first validator node.
+   Use the following command to copy the example configuration file to the
+   config directory, ``/etc/sawtooth``.
+
+   .. code-block:: console
+
+      user@client-1$ sudo cp /etc/sawtooth/validator.toml.example /etc/sawtooth/validator.toml
+
+   .. note::
+
+      This environment uses the default location and name for this configuration
+      file. For more information, see
+      :doc:`../sysadmin_guide/configuring_sawtooth/validator_configuration_file`.
+
+#. Edit ``/etc/sawtooth/validator.toml`` to change the following settings:
+
+   a. Set the ``bind`` directive to the network and component values for this
+      node.
+
+      * ``network`` specifies the validator-to-validator connection (by default,
+        ``127.0.0.1:8800``)
+
+      * ``component`` specifies the port on which the validator listens for
+        communication from other Sawtooth components (the REST API and
+        transaction processors)
+
+      For this application development environment, use the default settings:
+
+        .. code-block:: none
+
+           bind = [
+             "network:tcp://127.0.0.1:8800",
+             "component:tcp://127.0.0.1:4004"
+           ]
+
+   #. Set ``endpoint`` to the publicly addressable IP and port of the
+      validator-to-validator connection. By default, ``endpoint`` is the same as
+      the bind ``network`` setting: ``tcp://127.0.0.1:8800``. You must change
+      the ``endpoint`` value if the validator is hosted behind a firewall. In
```

Comment:
127.0.0.1 will never work for endpoint. each computer sees that address as it's own loopback interface. I wonder if we can be more clear here? Also while I did mention you need to change endpoint for NAT'ing / Firewall if the `network` is bound to anything but a routable IP address (i.e. if it is bound to 0.0.0.0 or 127.0.0.1) endpoint needs to list the actual IP of the machine (or the IP that the firewall is advertising). This is a tricky thing to communicate without a ton of words. 
Consider just conveying that the endpoint is the routable address and suggest there's a special case for a firewalled environment and maybe we can foot note that?
Or if we can just default the 'network' to 0.0.0.0 then maybe we can skip that part of the description and just pare back to this setting ('endpoint') and the 'peers' list. 
We can have an advanced network guide in the future to help walk people through the difference between bind and endpoint.

Reply:
See the above comment on why we can't use 0.0.0.0 as the default